### PR TITLE
21528-Improving-the-Quality-of-the-Bootstrap-process

### PR DESCRIPTION
**Pull request by tesonep**
Still Working On!

The idea of this issue is to improve the whole quality of the bootstrap process. 
Basically improve its modularity, its speed and documentation.

Initially it has:

  - Removed unused math methods from numbers and included in a new package loaded with hermes.
  - Removing debugging code from the bootstrap into a new package to be loaded with hermes.
  - Expressing the process in pharo code (minimizing the need for bash scripts).

Next it will have:
  - Minimizing the need to export things in .st format.
  - Using Mike to orchestrating the process.
  - Reducing initialization scripts and redundant steps.

The objectives are to be able to have an easier process, a more documented one, easily reproducible and using all we can inside pharo, and of course.... not needing to execute all the steps at once when running it locally.

Issue: https://pharo.fogbugz.com/f/cases/21528/Improving-the-Quality-of-the-Bootstrap-process

### CODE REVIEW

File: src/Deprecated70/Integer.extension.st [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	k := 1.
+
+	q := self - 1 bitShift: -1.
+	[q odd] whileFalse: 
+			[q := q bitShift: -1.
+			k := k + 1].
+	
+	25 timesRepeat: [(self isProbablyPrimeWithK: k andQ: q) ifFalse: [^false]].
+	^true
+]
```
Comment (jecisc) [inline src/Deprecated70/Integer.extension.st]: Maybe it should have a deprecation warning too.

**Review by jecisc — CHANGES_REQUESTED** [review]

File: src/Deprecated70/Integer.extension.st (hunk 1 above)
Comment (tesonep) [reply src/Deprecated70/Integer.extension.st]: Maybe we can even remove it directly, in the current image there is no implementation of the sent messages, so this isMostLikelyPrime produces a DNU. Also it is no used anywhere.

Comment (jecisc) [reply src/Deprecated70/Integer.extension.st]: If it produce a DNU then yes, maybe it's better to just remove it :)

**Comment by dionisiydk** [comment]
#'*Numbers-Extensions'  is not good name. It do not provide any meaning of the code inside. 
Better would be to call it Math-Extensions. Or "Numbers-Extended Math" (but package includes not only number subclass extensions)

**Comment by dionisiydk** [comment]
Another question does not it make worse system discovery? Because we can't categorise extension methods their visibility in browser will be changed.

Maybe such changes require support for protocol of method extensions.

**Comment by tesonep** [comment]
Yes, @dionisiydk I share your point (I will go with Math-Operations-Extensions), the problem with losing the categories is real. We should implement a better solution for extension methods, even being able to modify the structure of the extended class. In this case, most of the methods that are now extensions were in the "math functions" category, so if the extension is well named it has the same kind of information.

**Comment by dionisiydk** [comment]
I hope we can do it without hackish naming convention.
We need explicit package property for methods. I tried few times work on it https://pharo.fogbugz.com/f/cases/20887/CompiledMethod-package-should-not-rely-on-global-PackageOrganizer. But it was not easy and I put it for later.

**Review by pavel-krivanek — APPROVED** [review]

**Pull request merged**